include/drivers/virtualization/ivshmem.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_VIRTUALIZATION_IVSHMEM_H_
+#define ZEPHYR_INCLUDE_DRIVERS_VIRTUALIZATION_IVSHMEM_H_
+
+/**
+ * @brief ivshmem reference API
+ * @defgroup ivshmem ivshmem reference API
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef size_t (*ivshmem_get_mem_f)(const struct device *dev,
```

Comment:
Just wondering about the typedefs and naming. At least in networking side, the callback typedefs have been called `..._cb_t`, is this a new zephyr convention to use `.._f`?

Reply:
..._cb_t would be meaningful for an actual callback, ivshmem does not expose any.
That said there is no really naming convention on these specific types (which are only made to describe the API structure attributes). Some do not put anything at the end, some put _fn, some _f...